source/rate-limit.ts

```diff
@@ -459,9 +462,19 @@ const rateLimit = (
 						if (!(await config.requestWasSuccessful(request, response)))
 							await decrementKey()
 					})
+
+					// NOTE: A test in library/middleware-test.ts tests this, but it was
+					// disabled for being too flaky.
+					/* istanbul ignore next */
 					response.on('close', async () => {
 						if (!response.writableEnded) await decrementKey()
 					})
+
+					// NOTE: this may not be useful. None of the tests can trigger this
+					// callback (see `/crash` endpoint in test/library/helpers/create-server).
+					// Perhaps it is similar to the case described in this issue comment:
+					// https://github.com/nodejs/node/issues/44884#issuecomment-1270968365
```

Comment:
Good catch, that might well be the problem. Could be the cause of the flakiness on the 'close' event test also (?)

Reply:
OK, I think there's enough going on in this PR already, but maybe we can come back to it in the near future.